File: frontend/src/types/Product.ts
export interface Product {
  product_id: number;
  product_name: string;
  model?: string;
  description?: string;
  technical_specs?: string | Record<string, any>; // Allow either JSON string or object
  certifications?: string;
  safety_standards?: string;
  warranty_period: number;
  price: number;
  created_at: string;
  updated_at: string;
}

export interface ProductCreate extends Omit<Product, 'product_id' | 'created_at' | 'updated_at'> {
  product_id?: number;
  created_at?: string; // Optional for create payloads
  updated_at?: string; // Optional for create payloads
}

export interface ProductUpdate extends Omit<ProductCreate, 'product_id'> {
  product_id: number;
  created_at?: string; // Optional for update payloads
  updated_at?: string; // Optional for update payloads
} 

File: backend/internal/models/product.go
package models

import (
	"encoding/json"
	"time"
)

// Product maintains equipment details
type Product struct {
	ProductID       int             `db:"product_id" json:"product_id"`
	ProductName     string          `db:"product_name" json:"product_name"`
	Model           *string         `db:"model" json:"model,omitempty"`
	Description     *string         `db:"description" json:"description,omitempty"`
	TechnicalSpecs  json.RawMessage `db:"technical_specs" json:"technical_specs,omitempty"`
	Certifications  *string         `db:"certifications" json:"certifications,omitempty"`
	SafetyStandards *string         `db:"safety_standards" json:"safety_standards,omitempty"`
	WarrantyPeriod  int             `db:"warranty_period" json:"warranty_period"`
	Price           float64         `db:"price" json:"price"`
	CreatedAt       time.Time       `db:"created_at" json:"created_at"`
	UpdatedAt       time.Time       `db:"updated_at" json:"updated_at"`
}
